### PR TITLE
Adding converter for all properties of given type

### LINKED ISSUE #21: Support converters for all properties of specified type
Now if I want to use the same converter for several properties, I have to annotate them all. It would be nice to have a converter attribute for table, which will be used for all properties of specified type.

For now, the only useful example I have is, that I want for all string properties to convert `null` values to empty string (because string columns in database are `NOT NULL)`. And to have just one attribute on class is cleaner than to have attributed ever property.

We have to take care about:

- If the property itself has `Converter` attribute it has precedence over attribute on class.
- We need to say somehow that the property does not want to use converter from class (`NoConverterAttribute`?). For example if I have 10 string properties and I want to use converter for all of them except one.

### PR DESCRIPTION
**Pull request by satano**
Closes #21 

- [x] Reimplement existing fluent API using interfaces, so at any given moment, only allowed actions are possible.
- [x] Implement adding converter for all properties of given type (#21).
- [x] Document interfaces.
- [x] Move strings to resources.
- [x] Check existing tests and add new if it makes sense:
  - For exception when mapping the same property second time.
  - For new functionality.
- [x] Readme

### CODE REVIEW

File: src/Metadata/FluentConfiguration/EntityTypeBuilder.cs [hunk 1]
```diff
@@ -60,40 +60,62 @@ public EntityTypeBuilder<TEntity> HasTableName(string tableName)
...
         {
+            Check.NotNull(propertyExpression, nameof(propertyExpression));
             string propertyName = PropertyName<TEntity>.GetPropertyName(propertyExpression);
-
-            if (!_propertyBuilders.TryGetValue(propertyName, out PropertyBuilder<TEntity> propertyBuilder))
+            if (_propertyBuilders.ContainsKey(propertyName))
             {
-                propertyBuilder = new PropertyBuilder<TEntity>(this, propertyName);
-                _propertyBuilders.Add(propertyName, propertyBuilder);
+                throw new InvalidOperationException(string.Format("Property \"{0}\" was already configured.", propertyName));
```
Comment (Burgyn) [inline src/Metadata/FluentConfiguration/EntityTypeBuilder.cs]: Is necessary throws exception?
What when user try something like:
```csharp
modelBuilder.Entity<Foo>().Property(p=> p.Property1).HasColumnName("Name");
modelBuilder.Entity<Foo>().Property(p=> p.Property1).UseConverter<AddressConverter>();
```

File: src/Converter/NoConverter.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Kros.KORM.Converter
+{
+    internal class NoConverter : IConverter
+    {
+        public static IConverter Instance { get; } = new NoConverter();
```
Comment (Burgyn) [inline src/Converter/NoConverter.cs]: What about a private constructor? Force the use of default instance.

Comment (satano) [reply src/Converter/NoConverter.cs]: Maybe. But the class is internal anyway.

File: src/Metadata/FluentConfiguration/EntityTypeBuilder.cs (hunk 1 above)
Comment (satano) [reply src/Metadata/FluentConfiguration/EntityTypeBuilder.cs]: Yes. I was thinking about it. But I want to force to use "one-line" setup for properties. That way it is not possible to make configuration, which does not make sense. For example:

``` csharp
modelBuilder.Entity<Foo>().Property(p=> p.Property1).UseConverter<AddressConverter>();
modelBuilder.Entity<Foo>().Property(p=> p.Property1).IgnoreConverter();
modelBuilder.Entity<Foo>().Property(p=> p.Property1).NoMap();
```

File: src/Metadata/ConventionModelMapper.cs [hunk 3]
```diff
@@ -257,34 +230,33 @@ private ColumnInfo CreateColumnInfo(PropertyInfo propertyInfo, Type modelType)
 
         private void SetConverter(PropertyInfo propertyInfo, ColumnInfo columnInfo, Type modelType)
         {
-            IConverter converter = GetConverter(propertyInfo);
-
-            if (_converters.ContainsKey(modelType) && _converters[modelType].ContainsKey(columnInfo.PropertyInfo.Name))
+            if (_entities.TryGetValue(modelType, out EntityMapper entity))
             {
-                converter = _converters[modelType][columnInfo.PropertyInfo.Name];
+                if (entity.Converters.TryGetValue(columnInfo.PropertyInfo.Name, out IConverter converter)
```
[the remark this answers is not in the document]
Comment (Burgyn) [reply src/Metadata/ConventionModelMapper.cs]: Ok, By the tests I see that it works as it should, but I don't understand how it is possible.

File: src/Metadata/FluentConfiguration/EntityTypeBuilder.cs [hunk 4]
```diff
@@ -1,99 +1,119 @@
...
             {
-                modelMapper.SetInjector<TEntity>(_injector.Value);
+                modelMapper.SetConverterForProperties<TEntity>(item.Key, item.Value);
             }
 
+            var injectionConfig = new Lazy<InjectionConfiguration<TEntity>>(() => new InjectionConfiguration<TEntity>());
             foreach (PropertyBuilder<TEntity> propertyBuilder in _propertyBuilders.Values)
             {
-                propertyBuilder.Build(modelMapper);
+                if (propertyBuilder.IsMapped)
```
Comment (Burgyn) [inline src/Metadata/FluentConfiguration/EntityTypeBuilder.cs]: Why did you remove the Build method and replace it with several internal properties?
